auth/oidc_provider_test.go

```diff
@@ -41,75 +41,63 @@ func TestParseURIWithValidURL(t *testing.T) {
 	assert.Equal(t, httpsPort, url.Port())
 }
 
-// Check parsing an empty URL instead of a valid HTTP/HTTPS URL.
-// It should not return any URL or Error message.
-func TestParseURIWithEmptyURL(t *testing.T) {
+func TestParseURIWithBadURIs(t *testing.T) {
 	config := &OidcProviderConfiguration{}
-	url, err := config.parseURI("")
-	assert.Nil(t, url)
-	assert.Nil(t, err)
-}
-
-// Check parsing a non HTTP/HTTPS URL. It should should return an error
-// stating "Invalid URI scheme".
-func TestParseURIWithNonHttpURL(t *testing.T) {
-	config := &OidcProviderConfiguration{}
-	url, err := config.parseURI("sftp://accounts.unknown.com")
-	assert.Nil(t, url)
-	assert.Error(t, err)
-	assert.Contains(t, err.Error(), "Invalid URI scheme")
-}
-
-// Check parsing an HTTP/HTTPS URL with no Host. It should should return
-// an error stating "Host required in URI".
-func TestParseURIWithNoHost(t *testing.T) {
-	config := &OidcProviderConfiguration{}
-	url, err := config.parseURI("http://") // HTTP URL with No hostname
-	assert.Nil(t, url)
-	assert.Error(t, err)
-	assert.Contains(t, err.Error(), "Host required in URI")
-
-	url, err = config.parseURI("https://") // HTTPS URL with No hostname
-	assert.Nil(t, url)
-	assert.Error(t, err)
-	assert.Contains(t, err.Error(), "Host required in URI")
-}
+	tests := []struct {
+		input string
+		text  string
```

Comment:
rename to be something like `expectedErr` to make it clear it's used for checking the returned error

```suggestion
		expectedErr string
```

Reply:
Yeah, this has been done.